app/models/concerns/products/scheduling_support.rb

```diff
@@ -41,6 +41,35 @@ def next_available_reservation(after: Time.zone.now, duration: 1.minute, options
     reservation_in_week(after, duration, rules, options)
   end
 
+  def quick_reservation_data(after: Time.zone.now + 1.minute)
+    quick_reservation_reservations(after:).map { |r| { reserve_start_at: r.reserve_start_at, duration_mins: r.duration_mins } }
```

Comment:
Would this work instead of relying on `quick_reservation_reservations`?
```suggestion
    quick_reservation_intervals.select { |i| walkup_available?(after, interval: i) }.compact
```